subsys/bluetooth/host/adv.c

```diff
@@ -524,27 +524,34 @@ static int hci_set_ad(uint16_t hci_op, const struct bt_ad *ad, size_t ad_len)
 }
 
 static int hci_set_adv_ext_complete(struct bt_le_ext_adv *adv, uint16_t hci_op,
-				    const struct bt_ad *ad, size_t ad_len)
+				    size_t total_len, const struct bt_ad *ad, size_t ad_len)
 {
 	struct bt_hci_cp_le_set_ext_adv_data *set_data;
 	struct net_buf *buf;
 	int err;
 	uint8_t max_data_size;
 
-	buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
-	if (!buf) {
-		return -ENOBUFS;
-	}
-
-	set_data = net_buf_add(buf, sizeof(*set_data));
-	(void)memset(set_data, 0, sizeof(*set_data));
-
 	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
 		max_data_size = BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN;
 	} else {
 		max_data_size = BT_GAP_ADV_MAX_ADV_DATA_LEN;
 	}
 
+	if (total_len > max_data_size) {
+		return -EINVAL;
+	}
```

Comment:
Since we already check `if (total_len_bytes <= BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN)` before calling this function, this check could be move into the non-ext `else` clause.